vic/drivers/shared_image/src/state_metadata.c

```diff
@@ -0,0 +1,676 @@
+           "STATE_LAKE_ICE_AREA");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC].long_name, "lake_ice_area");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC].standard_name,
+           "lake_ice_coverage");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC].units, "m2");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC].description,
+           "area of ice coverage on lake at beginning of time step");
+
+    // STATE_LAKE_ICE_AREA_FRAC_NEW
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC_NEW].varname,
+           "STATE_LAKE_ICE_AREA_NEW");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC_NEW].long_name,
+           "lake_ice_area_new");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC_NEW].standard_name,
+           "lake_ice_area_new");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC_NEW].units, "m2");
+    strcpy(state_metadata[STATE_LAKE_ICE_AREA_FRAC_NEW].description,
+           "area of ice coverage on lake at end of time step");
+
+    // STATE_LAKE_ICE_WATER_EQUIVALENT
+    strcpy(state_metadata[STATE_LAKE_ICE_WATER_EQUIVALENT].varname,
+           "STATE_LAKE_ICE_WATER_EQUIVALENT");
+    strcpy(state_metadata[STATE_LAKE_ICE_WATER_EQUIVALENT].long_name,
+           "lake_ice_water_equivalent");
+    strcpy(state_metadata[STATE_LAKE_ICE_WATER_EQUIVALENT].standard_name,
+           "lake_ice_water_equivalent");
+    strcpy(state_metadata[STATE_LAKE_ICE_WATER_EQUIVALENT].units, "m3");
+    strcpy(state_metadata[STATE_LAKE_ICE_WATER_EQUIVALENT].description,
+           "liquid water equivalent volume of lake ice");
+
+    // STATE_LAKE_ICE_HEIGHT
+    strcpy(state_metadata[STATE_LAKE_ICE_HEIGHT].varname,
+           "STATE_LAKE_ICE_HEIGHT");
+    strcpy(state_metadata[STATE_LAKE_ICE_HEIGHT].long_name, "lake_ice_height");
+    strcpy(state_metadata[STATE_LAKE_ICE_HEIGHT].standard_name,
+           "lake_ice_height_thickest");
+    strcpy(state_metadata[STATE_LAKE_ICE_HEIGHT].units, "m");
+    strcpy(state_metadata[STATE_LAKE_ICE_HEIGHT].description,
+           "lake ice height at ghickest point");
+
+    // STATE_LAKE_ICE_TEMP
+    strcpy(state_metadata[STATE_LAKE_ICE_TEMP].varname, "STATE_LAKE_ICE_TEMP");
+    strcpy(state_metadata[STATE_LAKE_ICE_TEMP].long_name, "lake_ice_temp");
+    strcpy(state_metadata[STATE_LAKE_ICE_TEMP].standard_name,
+           "lake_ice_temperature");
+    strcpy(state_metadata[STATE_LAKE_ICE_TEMP].units, "C");
+    strcpy(state_metadata[STATE_LAKE_ICE_TEMP].description,
+           "lake ice temperature");
+
+    // STATE_LAKE_ICE_SWE
+    strcpy(state_metadata[STATE_LAKE_ICE_SWE].varname, "STATE_LAKE_ICE_SWE");
+    strcpy(state_metadata[STATE_LAKE_ICE_SWE].long_name, "lake_ice_swe");
+    strcpy(state_metadata[STATE_LAKE_ICE_SWE].standard_name,
+           "lake_snow_water_equivalent");
+    strcpy(state_metadata[STATE_LAKE_ICE_SWE].units, "m");
+    strcpy(state_metadata[STATE_LAKE_ICE_SWE].description,
+           "liquid water equivalent depth of lake snow");
+
```

Comment:
I'll look at the lake state section when I do my review.  But I think we should make lake state a separate issue from put_data.  I don't know if you all think that sorting out the lake states (and reading/writing only the minimum necessary variables) should be taken care of before releasing 5.0.  If you do, then I can work on that.  But either way, it seems separate from, and shouldn't hold up, finalizing the put_data refactor.